integration_tests/src/main/python/array_test.py

```diff
@@ -292,34 +293,37 @@ def test_array_slice(data_gen):
             'slice(array(array(1, null, 2), array(1), array(null)), NULL, NULL)'))
 
 
-@pytest.mark.parametrize('data_gen', array_item_test_gens, ids=idfn)
-def test_array_slice_with_illegal_arguments(data_gen):
-    length_gen = IntegerGen(min_val=0, max_val=100, special_cases=[None])
-    neg_length_gen = IntegerGen(min_val=-25, max_val=-1, special_cases=[None])
+@pytest.mark.parametrize('zero_start', [0, 'b'], ids=idfn)
+@pytest.mark.parametrize('valid_length', [5, 'c'], ids=idfn)
+@pytest.mark.parametrize('data_gen', array_item_test_gens_sans_allnull, ids=idfn)
+def test_array_slice_with_zero_start(data_gen, zero_start, valid_length):
+    zero_start_gen = IntegerGen(nullable=False, min_val=0, max_val=0, special_cases=[])
+    valid_length_gen = IntegerGen(min_val=0, max_val=100, special_cases=[None])
+    # When the list column is all null, the result is also all null regardless of the start and length
+    assert_gpu_and_cpu_are_equal_collect(
```

Comment:
Nit: it is a valid when list column is all null, this test case is testing invalid cases, so it's better move the valid case to test_array_slice